ports/stm32/uart.c

```diff
@@ -717,6 +717,20 @@ void uart_irq_handler(mp_uint_t uart_id) {
         return;
     }
 
+    // Set user IRQ flags
+    #if defined(STM32F4)
+    self->mp_irq_flags = self->uartx->SR;
+    if (self->uartx->SR & USART_SR_IDLE) {
+        (void)self->uartx->SR;
+        (void)self->uartx->DR;
```

Comment:
I'm pretty sure this will break the RXNE handler below because it'll read the current incoming data (from DR) and clear the RXNE flag.  Probably best to cache the SR and DR values here, or similar.  It's a bit tricky, and is different on F4 vs other MCUs.

Reply:
Do you know why DR was ever being read/cleared by this block? 

The way I read it, if the rx buffer was already full (above) DR would not have been read into the buffer, but then this would clear / throw it away. Perhaps it's better to just clear SR and delete the DR line.

Reply:
ok ignore that, the ref manual does list this as the only way to clean idle flag.